features/netsocket/nsapi_types.h

```diff
@@ -266,6 +268,7 @@ typedef enum nsapi_socket_option {
     NSAPI_ADD_MEMBERSHIP,    /*!< Add membership to multicast address */
     NSAPI_DROP_MEMBERSHIP,   /*!< Drop membership to multicast address */
     NSAPI_BIND_TO_DEVICE,        /*!< Bind socket network interface name*/
+    NSAPI_IP_TOS,            /*!< IP TOS */
```

Comment:
Separate out from this PR - if you're adding it, I would want to see at least 1 of lwIP or Nanostack support to illustrate the parameter type.